specs/derivation.md

```diff
@@ -0,0 +1,537 @@
+to ensure that if the sequencing window for a L2 block has changed since it was derived,
+that L2 block is re-derived.
+
+The first L1 block of the sequencing window contains the L1 attributes for that L2 block.
+The end of the sequencing window is the canonical L1 block whose number is `SEQUENCING_WINDOW` larger than the start.
+The end of the window must be selected by number otherwise the sequencer would not be able to create batches.
+The problem with selecting the end of the window by number is that when an L1 reorg occurs,
+the blocks (and thus batches) in the window could change.
+We must find the first L2 block whose complete sequencing window is unchanged in the reorg.
+
+[merge]: https://ethereum.org/en/eth2/merge/
+
+#### Resetting the Channel Bank
+
+In the case of the Channel Bank the origin has to be reset further back,
+by a full `CHANNEL_TIMEOUT` behind the desired L1 origin to continue the pipeline from.
+The pipeline will drop the outputs from the Channel Bank stage until it has caught up to the next origins.
+
+This ensures the reset is performed in the same pipeline of small recoverable specs, without additional complexity.
+
+The rollback by a full `CHANNEL_TIMEOUT` is required to avoid previous channel frames from missing in the Channel Bank,
+when it continues from the new safe L1 origin that was selected after the reorg.
+
+#### Reorgs Post-Merge
+
+Note that post-[merge], the depth of re-orgs will be bounded by the [L1 finality delay][l1-finality]
+(every 2 epochs in the happy case, approximately 12 minutes).
+
+(\*) Post-merge, this is only possible for 12 minutes. In practice, we'll pick an already-finalized L1 block as L2
+inception point to preclude the possibility of a re-org past genesis, at the cost of a few empty blocks at the start of
+the L2 chain.
+
+
+### L1 Data Source
+
+The first stage starts at a L1 origin, and moves the next L1 origin on a read.
+If there is no next L1 origin yet, the stage returns an `EOF` like an intermediate stage would, to indicate all input was exhausted,
+and the next pipeline step may need to be triggered only after a hint of new inputs (e.g. the L1 head changed).
+
+After retrieving the next L1 origin, the data for the origin is retrieved from a [Data Availability Source](#data-availability-source).
+
+#### Data Availability Source
+
+A data availability source fetches a list of bytestrings corresponding to a certain L1 block hash/number combination.
+
+By default, ethereum calldata is supported as data-availability source.
+
+Transactions of the queried block are traversed, and filtered by receiver and sender:
+- The transaction receiver MUST be the sequencer inbox address
+- The transaction must be signed by a recognized batch submitter account
+
+
+### ChannelBank
+
+#### Constants
+
+```text
+CHANNEL_TIMEOUT = 600   # time in seconds until a channel is pruned from the channel bank
+MAX_CHANNEL_BANK_SIZE = 100_000_000  # max size of the combined channels in the channel bank
```

Comment:
I think a per channel max size + cap on the number of open channels is more resilient.